### PR TITLE
feat(api/antivirus): secret api key for antivirus

### PR DESCRIPTION
**Pull request by michaelgarcia75**
**Pull Request Title Rules**

Pull request format must respect this format:

```
<type>(<scope>, <scope>...): <Notion ID> or "sentry" - <description>
```

Allowed types: `feat`, `fix`, `refactor`, `chore`.
Allowed scopes: `release`, `app`, `admin`, `misc`, `api`, `all`, `github`, `terraform`, `kb`, `lib`.

Notion ID, followed by a `-` is **mandatory** on types `refactor`, `feat` and `fix`.

_If you really need to bypass the pull request title validation, you can add the `bypass title` label to you PR._

Examples of valid PR titles:

- `chore(release): New version 1.3.3.7`
- `refactor(api): 1789 - Removed monarchy from codebase`
- `fix(api, app, admin): 1981 - User cannot delete each other accounts anymore.`
- `fix(api): sentry - Implement capture or fix a bug.`

**Description**

**Todo**

**Checklist**

- [ ] **⚠️ My code can have side-effects on other part of the code-base**
- [ ] I have added the Plausible tags/events
- [ ] I have performed a self-review of my code (and removed console.log)

**Ticket / Issue**

**Testing instructions**

### CODE REVIEW

File: .github/workflows/cc-deploy-production.yml [hunk 1]
```diff
@@ -15,7 +15,6 @@ on:
         type: string
```
Comment (pitiscarf) [inline .github/workflows/cc-deploy-production.yml:15]: Il faut rajouter antivirus_image_tag comme variable d'entrée :

`antivirus_image_tag:
    required: true
    type: string
`

File: .github/workflows/cc-sync-production.yml [hunk 2]
```diff
@@ -51,6 +52,10 @@ jobs:
           echo "app_image_tag: $app_image_tag"
           echo "app_image_tag=$app_image_tag" >> $GITHUB_OUTPUT
 
+          antivirus_image_tag=$(jq -r '.antivirus_image_tag.value' <<< $output)
+          echo "antivirus_image_tag: $antivirus_image_tag"
+          echo "antivirus_image_tag=$antivirus_image_tag" >> $GITHUB_OUTPUT
+
   deploy:
     needs: prepare
     uses: ./.github/workflows/cc-deploy-production.yml
```
Comment (pitiscarf) [inline .github/workflows/cc-sync-production.yml:61]: rajouter `antivirus_image_tag: ${{ needs.prepare.outputs.antivirus_image_tag }}`

File: api/src/utils/virusScanner.js [hunk 3]
```diff
@@ -17,12 +37,13 @@ async function scanFile(tempFilePath, name, userId = "anonymous") {
   const scan = async () => {
     const stream = createReadStream(tempFilePath);
     const url = `${config.API_ANTIVIRUS_ENDPOINT}/scan`;
-
+    const token = await getAccessToken(config.API_ANTIVIRUS_ENDPOINT, config.API_ANTIVIRUS_KEY);
```
Comment (pitiscarf) [inline api/src/utils/virusScanner.js]: pourquoi ne pas recuperer le token avant createReadStream ?

File: api/src/utils/virusScanner.js [hunk 4]
```diff
@@ -17,12 +37,13 @@ async function scanFile(tempFilePath, name, userId = "anonymous") {
...
     const formData = new FormData();
     formData.append("file", stream);
 
     const headers = formData.getHeaders();
     headers["X-Auth-Token"] = config.API_ANTIVIRUS_TOKEN;
```
Comment (pitiscarf) [inline api/src/utils/virusScanner.js]: config.API_ANTIVIRUS_TOKEN n'est plus utile. Il faut le supprimer de la config et des secrets

**Review by pitiscarf — CHANGES_REQUESTED** [review]

File: api/src/utils/virusScanner.js (hunk 4 above)
Comment (michaelgarcia75) [reply api/src/utils/virusScanner.js]: ah ok j'étais pas sur donc je l'ai laissé au cas ou

File: api/src/utils/virusScanner.js (hunk 3 above)
Comment (michaelgarcia75) [reply api/src/utils/virusScanner.js]: Aucune raison particulière mais oui ça aurait été mieux de le mettre au début

File: .github/workflows/cc-sync-production.yml (hunk 2 above)
Comment (michaelgarcia75) [reply .github/workflows/cc-sync-production.yml:61]: Bien vu j'étais pas sûr de rien avoir loupé

**Review by pitiscarf — APPROVED** [review]

**Pull request merged**